### ISSUE TITLE
Notification Action Button

### ISSUE THREAD

**Issue by stipe76** [home-assistant/android#1163]
**Is your feature request related to a problem? Please describe.**

No

**Describe the solution you'd like**

Is it possible to call a service from an action button in an actionable notification? I'd like to be able to fire off a webhook directly on button button press. Alternatively, is it possible for the URI action to allow HTTP POST and JSON data?

**Describe alternatives you've considered, if any**

**Additional context**

**Comment by dshokouhi** [comment]
You should be reacting to the action events in your automations as part of the actionable automation workflow.  The event gets fired immediately to your HA server to process.

**Comment by stipe76** [comment]
> You should be reacting to the action events in your automations as part of the actionable automation workflow. The event gets fired immediately to your HA server to process.

Yep, that's what I'm currently doing. I am also using Tasker to send a webhook from the Notification Bar, and it is so much faster firing a direct HTTP POST than HA reacting to an event. I can live with it, but it would be nice to have this as an option. 

Cheers

**Comment by dshokouhi** [comment]
Would this be the same as #396 ?

**Comment by stipe76** [comment]
> Would this be the same as #396 ?

In part, yes. Something along these lines would be nice

`service: notify.mobile_app_pixel_4
   data:
      message: "The light has been left on"
      data:
         actions:
            - action: URI_POST
              title: "Turn off light"
              uri: "http://webhook.url"
              action_data:
                 my_custom_data: {"what":"ever"}`

Or similar

**Comment by royto** [comment]
I'm also interested in having possibilities to get actions_data implemented in Android.

Could simplify a lot my automations

**Comment by stipe76** [comment]
Any updates on this?

**Comment by homeoneautomacao** [comment]
Hi, when I use the code below, it works normally. When I click on the OPEN CAMERA notification button it opens the app and opens the camera.garage entity. But when I change my user settings, changing the "default" dashboard to any other dashboard, it doesn't work. When I click on OPEN CAMERA it opens the selected dashboard and not the camera.garage entity. I tested it with other entities like sun.sun and the error is the same.

service: notify.mobile_app_mi_9t
data:
  message: The garage door has been open for 10 minutes.
  title: Your Garage Door Friend
  data:
    actions:
      - action: URI
        title: OPEN CAMERA
        uri: entityId:camera.garage

Thanks

**Comment by dshokouhi** [comment]
@homeoneautomacao please file a separate issue as this is a feature request, make sure to be very descriptive in the bug that you create
